packages/e2e/lib/common.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+/// An object sent from e2e back to the Flutter Driver in response to
+/// `request_data` command.
+class Response {
+  String _failureDetails;
+
+  final bool _allTestsPassed;
+
+  /// Constructor which receives tests results' flag and failure details.
+  Response(this._allTestsPassed, String failureDetails) {
+    this._failureDetails = failureDetails;
+  }
+
+  /// Whether the test run successfully or not.
+  String get result => _allTestsPassed ? 'pass' : 'fail';
+
+  /// If the result are failures get the formatted details.
+  String get failureDetails => _allTestsPassed ? '' : _failureDetails;
+
+  /// Convert a string pass/fail result to a boolean.
+  static bool allTestsPassed(String result) => (result == 'pass')
```

Comment:
Does this need to be public?

Reply:
I am providing as many public methods as possible.

I think you already noticed. This is a breaking change. There are many tests already written using e2e. I am trying to provide methods to developers:
- a method where they can call if they want to keep their string usage ('pass', 'fail')
- a method where they can call for a boolean.